src/ui/public/test_harness/test_sharding/setup_test_sharding.js

```diff
@@ -0,0 +1,53 @@
+import { uniq } from 'lodash';
+
+import { findBundleUrl } from './find_bundle_url';
+import { getShardingParamsFromUrls } from './get_sharding_params_from_urls';
+import { shimSuiteCreation } from './shim_suite_creation';
+import { getShardNum } from './get_shard_num';
+
+export function setupTestSharding() {
+  // read config from the page url, and from the bundle url, so that we can
+  // override params in dev mode
+  const params = getShardingParamsFromUrls([
```

Comment:
Let's move `DEFAULT_PARAMS` and the logic from `getShardingParamsFromUrls` in here, e.g.

```javascript
const windowUrlShardingParams = getShardingParamsFromUrl(window.location.href);
const testBundleShardingParams = getShardingParamsFromUrl(findTestBundleUrl());
const shardingParams = defaults({}, windowUrlShardingParams, testBundleShardingParams, DEFAULT_PARAMS);
```